pkg/generator/main.go

```diff
@@ -0,0 +1,158 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package main
+
+import (
+	"fmt"
+	"log"
+	"os"
+	"regexp"
+	"strings"
+
+	apiext "k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/v1"
+	"sigs.k8s.io/controller-tools/pkg/crd"
+	"sigs.k8s.io/controller-tools/pkg/loader"
+	"sigs.k8s.io/controller-tools/pkg/markers"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	ExperimentalMarker      = "<gateway:experimental>"
+	BundleVersionAnnotation = "gateway.networking.k8s.io/bundle-version"
+	ChannelAnnotation       = "gateway.networking.k8s.io/channel"
+)
+
+// This generation code is largely copied from sigs.k8s.io/controller-tools/pkg/crd/gen.go
```

Comment:
Can you reference the commit so that we can navigate up and down the git history as that file changes upstream?

Reply:
Updated, thanks!